handshake_messages.go

```diff
@@ -1114,7 +1114,11 @@ func (m *serverHelloMsg) unmarshal(data []byte) alert {
 		if m.vers != VersionTLS12 {
 			return alertDecodeError
 		}
-		m.vers = uint16(svData[0])<<8 | uint16(svData[1])
+		rcvVer := binary.BigEndian.Uint16(svData[0:])
+		if rcvVer < VersionTLS13 {
```

Comment:
The client should also check that `rcVers` is among the versions it offered in its "supported_version" extension. Is this check made? If not, add it; if it is, I'd move this check to wherever that is.

Reply:
it's done